packages/libs/user-datasets/src/lib/Service/api.ts

```diff
@@ -100,6 +101,15 @@ export class UserDatasetApi extends FetchClientWithCredentials {
           console.error(error);
         }
       }
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status >= 400) {
+        try {
+          throw new Error(xhr.response);
+        } catch (error) {
+          dispatchUploadProgress && dispatchUploadProgress(null);
+          dispatchBadUpload && dispatchBadUpload(String(error));
+          console.log(error);
+        }
```

Comment:
I don't think you need the try/catch here. You should be able to just do

```suggestion
          const error = new Error(xhr.response);
          dispatchUploadProgress && dispatchUploadProgress(null);
          dispatchBadUpload && dispatchBadUpload(String(error));
          console.log(error);
```